recipes/opencv/4.x/conanfile.py

```diff
@@ -0,0 +1,311 @@
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and \
+           "MT" in str(self.settings.compiler.runtime) and self.options.shared:
+            raise ConanInvalidConfiguration("Visual Studio and Runtime MT is not supported for shared library.")
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build_requirements(self):
+        if self.settings.os == "Linux":
+            self.build_requires("gtk/system")
+
+    def system_requirements(self):
+        if tools.os_info.is_linux and self.settings.os == "Linux":
+            # TODO: Get this as a gtkmm/system package or integrated in gtk/system
+            gtkmm_version = 24
+            package_tool = tools.SystemPackageTool(conanfile=self, default_mode="verify")
+            if tools.os_info.with_apt:
+                packages = ["libgtkmm-2.4-dev"]
+            elif tools.os_info.with_yum or tools.os_info.with_dnf:
+                packages = ["gtkmm{}-devel".format(gtkmm_version)]
+            elif tools.os_info.with_pacman:
+                packages = ["gtkmm"]
+            elif tools.os_info.with_zypper:
+                packages = ["gtk{}-devel".format(gtkmm_version)]
+            else:
+                self.output.warn("Do not know how to install 'gtkmm-{}' for {}."
+                                 .format(gtkmm_version,
+                                         tools.os_info.linux_distro))
+                packages = []
+            for p in packages:
+                package_tool.install(update=True, packages=p)
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+        if self.options.with_jpeg:
+            self.requires("libjpeg/9d")
+        if self.options.with_png:
+            self.requires("libpng/1.6.37")
+        if self.options.with_jasper:
+            self.requires("jasper/2.0.21")
+        if self.options.with_openexr:
+            self.requires("openexr/2.5.3")
+        if self.options.with_tiff:
+            self.requires("libtiff/4.1.0")
+        if self.options.with_eigen:
+            self.requires("eigen/3.3.8")
+        if self.options.with_tbb:
+            self.requires("tbb/2020.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("opencv-{}".format(self.version), self._source_subfolder)
+
+    def _patch_opencv(self):
+        tools.rmdir(os.path.join(self._source_subfolder, "3rdparty"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions['CMAKE_CXX_STANDARD'] = '11'
```

Comment:
This looks a little suspicious =? Is this not managed by conan?

Reply:
It was originally there as a patch for openEXR in 2.4.13.7 that I rewrote. I will retest, it probably does not need to be there anymore.

Reply:
Ok, tested and it seems to be spurious indeed. I removed it.